plugins/@grouparoo/airtable/src/lib/client/client.ts

```diff
@@ -0,0 +1,188 @@
+    const filterString = `OR(${conditionals.join(",")})`;
+    return this.baseClient(tableId)
+      .select({
+        filterByFormula: filterString,
+        maxRecords: foreignKeys.length,
+      })
+      .all();
+  }
+
+  /**
+   * @throws AirtableError - Airtable will throw an error if there is any issue destroying the record
+   * @param tableId - unique identifier of the table
+   * @param recordId - unique record identifier within the table
+   * @returns Promise<Record<FieldSet>> - A destroyed record with only its id in the returned field set
+   */
+  async deleteRecord(
+    tableId: string,
+    recordId: string
+  ): Promise<Record<FieldSet>> {
+    return this.baseClient(tableId).destroy(recordId);
+  }
+  /**
+   * @throws AirtableError - Airtable will throw an error if there is any issue destroying the record
+   * @param tableId - unique identifier of the table
+   * @param recordIds - unique record identifiers within the table
+   * @returns Promise<Record<FieldSet>> - An array of destroyed records with only their id in the returned field set
+   */
+  async deleteRecords(
+    tableId: string,
+    recordIds: string[]
+  ): Promise<Records<FieldSet>> {
+    return this.baseClient(tableId).destroy(recordIds);
+  }
+
+  /**
+   * @throws AirtableError - Airtable will throw an error if there is any issue updating the record
+   * @param tableId - unique identifier of the table
+   * @param record - unique record identifier with the partial set of fields to update
+   * @returns Promise<Record<FieldSet>> -  record with its updated fields and other unchanged fields
+   */
+  async updateRecord(tableId: string, record: RecordData<Partial<any>>) {
+    return this.baseClient(tableId).update(record.id, record.fields);
+  }
+
+  /**
+   * @throws AirtableError - Airtable will throw an error if there is any issue destroying the record
+   * @param tableId - unique identifier of the table
+   * @param records - unique record identifiers with the partial set of fields to update
+   */
+  async updateRecords(
+    tableId: string,
+    records: RecordData<Partial<FieldSet>>[]
+  ): Promise<Records<FieldSet>> {
+    return this.baseClient(tableId).update(records);
+  }
+
+  /**
+   * Create a single record given a map of fields
+   * @param tableId - unique identifier of the table
+   * @param record - unique record fields
+   */
+  async createRecord(
+    tableId: string,
+    record: Partial<FieldSet>
+  ): Promise<Record<FieldSet>> {
+    return this.baseClient(tableId).create(record);
+  }
+
+  /**
+   * Create Multiple records
+   * @param tableId - unique identifier of the table
+   * @param records - unique record fields with updates to
+   */
+  async createRecords(tableId: string, records: CreateRecord<FieldSet>[]) {
+    return this.baseClient(tableId).create(records);
+  }
+
+  /**
+   * NOOP method for mocking out a clossing of the client
+   */
+  async close(): Promise<void> {
+    return Promise.resolve();
+  }
```

Comment:
You can skip the method if you want, and then not implement the disconnect method on the grouparoo plugin